mlflow/pyfunc/__init__.py

```diff
@@ -1097,6 +1110,7 @@ def load_model(
         model_impl=model_impl,
         predict_fn=predict_fn,
         predict_stream_fn=predict_stream_fn,
+        model_id=_parse_model_id_if_present(model_uri),
```

Comment:
Note that we'll need to update `_parse_model_id_if_present` once we also add a model ID to Registered Model Versions, since we'll want to fetch the model ID if the user loads the version via `mlflow.pyfunc.load_model("models:/myname/<version_number>")`. Can we file a JIRA for that so we remember in the future? cc @prithvikannan